models/repo/language_stats.go

```diff
@@ -43,29 +44,66 @@ func (stats LanguageStatList) LoadAttributes() {
 
 func (stats LanguageStatList) getLanguagePercentages() map[string]float32 {
 	langPerc := make(map[string]float32)
-	var otherPerc float32 = 100
+	var otherPerc float32
 	var total int64
 
 	for _, stat := range stats {
 		total += stat.Size
 	}
 	if total > 0 {
 		for _, stat := range stats {
-			perc := float32(math.Round(float64(stat.Size)/float64(total)*1000) / 10)
+			perc := float32(float64(stat.Size) / float64(total) * 100)
 			if perc <= 0.1 {
+				otherPerc += perc
 				continue
 			}
-			otherPerc -= perc
 			langPerc[stat.Language] = perc
 		}
-		otherPerc = float32(math.Round(float64(otherPerc)*10) / 10)
 	}
 	if otherPerc > 0 {
 		langPerc["other"] = otherPerc
 	}
+	roundByLargestRemainder(langPerc, 100)
 	return langPerc
 }
 
+// Rounds to 1 decimal point, target should be the expected sum of percs
+func roundByLargestRemainder(percs map[string]float32, target float32) {
+	// Intermediates are the floored numbers multiplied to the decimal
+	// 54.37 -> 543
+	intermediates := make(map[string]int, len(percs))
+	remainders := make(map[string]float64, len(percs))
```

Comment:
I think you can get away without these two maps by just using the percs map directly.

Reply:
You probably could but would need to loop through twice, once to sum up the int and the 2nd to += the highest fractional

Reply:
Thanks, I can see that I can store the remainders there.
I don't see how to avoid using the intermediates map.

Reply:
A floating point number can be thought of as two numbers in one, the integer part and the remainder.

You just keep both parts around, just use the integer part as intermediates and the remainder part as the remainder.